pkg/apis/core/validation/shoot.go

```diff
@@ -1307,6 +1307,9 @@ func ValidateKubeAPIServer(kubeAPIServer *core.KubeAPIServerConfig, version stri
 			if err != nil || (issuer != nil && len(issuer.Host) == 0) {
 				allErrs = append(allErrs, field.Invalid(oidcPath.Child("issuerURL"), oidc.IssuerURL, "must be a valid URL and have https scheme"))
 			}
+			if issuer != nil && issuer.Fragment != "" {
```

Comment:
A note:

Kubernetes also validate that RawQuery and User are empty https://github.com/kubernetes/kubernetes/blob/776fb24e066d54cf84a52366aeea121fa5748e27/staging/src/k8s.io/apiserver/pkg/apis/apiserver/validation/validation.go#L169-L180

The[ OIDC spec](https://openid.net/specs/openid-connect-core-1_0.html) says:
>  Issuer Identifier
Verifiable Identifier for an Issuer. An Issuer Identifier is a case-sensitive URL using the https scheme that contains scheme, host, and optionally, port number and path components and no query or fragment components.